From 3ddefd7e1fe7c40cd8d415c80fe3371c57b3498a Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Thu, 27 Aug 2015 20:34:32 +0800
Subject: [PATCH 0965/5242] MLK-11438-4 lcdif: enable lpsr mode

commit  c9db921d889dda34e7322cd64a508f20ca96d793 from
https://source.codeaurora.org/external/imx/linux-imx.git

restore the pinmux when resume from LPSR mode in suspend.

Signed-off-by: Robby Cai <r63905@freescale.com>
(cherry picked from commit 0462eb2cb8b64508260d9c6daa9f163d68ea6be0)

Conflicts:
	drivers/video/mxsfb.c

(cherry picked from commit 8ddaa850df8cc9fe4dda4f505053eaa3f704ef69)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 6a96806..389d504 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1565,6 +1565,9 @@ static int mxsfb_suspend(struct device *pdev)
 	mxsfb_blank(FB_BLANK_POWERDOWN, fb_info);
 	host->restore_blank = saved_blank;
 	console_unlock();
+
+	pinctrl_pm_select_sleep_state(pdev);
+
 	return 0;
 }
 
@@ -1573,6 +1576,8 @@ static int mxsfb_resume(struct device *pdev)
 	struct fb_info *fb_info = dev_get_drvdata(pdev);
 	struct mxsfb_info *host = fb_info->par;
 
+	pinctrl_pm_select_default_state(pdev);
+
 	console_lock();
 	mxsfb_blank(host->restore_blank, fb_info);
 	fb_set_suspend(fb_info, 0);
-- 
1.7.9.5

